### PR TITLE
RM-9206 Upgrade Microsoft.CodeAnalysis.CSharp.Workspaces to 4.9.2

### PR DESCRIPTION
**Pull request by KuJo7**
https://re-motion.atlassian.net/browse/RM-9206

### CODE REVIEW

File: Remotion/Core/Development.Analyzers/RMCORE0001_SafeContextAnalyzer.cs [hunk 1]
```diff
@@ -17,15 +17,15 @@ private record SymbolContext (ITypeSymbol? SafeContextBoundarySymbol);
     public static readonly DiagnosticDescriptor AlternativeDescriptor = new DiagnosticDescriptor(
         "RMCORE0001",
         "Use SafeContext instead of typical API",
-        "'{0}' should not be used with SafeContext, use 'Remotion.Context.SafeContext.{1}' instead.",
+        "'{0}' should not be used with SafeContext, use 'Remotion.Context.SafeContext.{1}' instead",
```
Comment (MichaelKetting) [inline Remotion/Core/Development.Analyzers/RMCORE0001_SafeContextAnalyzer.cs:20]: Micorsoft now uses a form without a trailing period in the analyzers? it would be best to have this as a separate commit with an explainatory commit message.

Context: I remember a discussion (or change) about this from 12-24 months ago.  I would like to understand when and how this was changed (or if we always did it "wrong").
cc @a-ctor & @Frazerus

File: Remotion/Web/Development.Analyzers/AnalyzerReleases.Unshipped.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+### New Rules
```
Comment (MichaelKetting) [inline Remotion/Web/Development.Analyzers/AnalyzerReleases.Unshipped.md:1]: @a-ctor  & @Frazerus There was a discussion about not needed this if I remember correctly. Apparently, this semantic has now changed. Thoughts?

File: Remotion/Core/Development.Analyzers/Core.Development.Analyzers.csproj [hunk 3]
```diff
@@ -7,6 +7,7 @@
     <AssemblyName>Remotion.Development.Analyzers</AssemblyName>
     <AssemblyTitle>Remotion Analyzers</AssemblyTitle>
     <Description>Analyzer ensuring best practices when using the Remotion assemblies.</Description>
+    <EnforceExtendedAnalyzerRules>true</EnforceExtendedAnalyzerRules>
```
Comment (MichaelKetting) [inline Remotion/Core/Development.Analyzers/Core.Development.Analyzers.csproj:10]: What does the new property do? what happens by setting it true (or false)?

File: Remotion/Web/Development.Analyzers/WxeResourcePageStepAnalyzer.cs [hunk 4]
```diff
@@ -93,7 +93,7 @@ private void AnalyzeObjectCreationExpression (SyntaxNodeAnalysisContext ctx, Sym
 
       if (operation.Arguments[0].Value is not ITypeOfOperation typeOfOperation)
       {
-        if (SymbolEqualityComparer.Default.Equals(operation.Arguments[0].Parameter.Type, symbolContext.SystemTypeSymbol))
+        if (SymbolEqualityComparer.Default.Equals(operation.Arguments[0].Parameter!.Type, symbolContext.SystemTypeSymbol))
```
Comment (MichaelKetting) [inline Remotion/Web/Development.Analyzers/WxeResourcePageStepAnalyzer.cs:96]: I didn't look into this beyond the code I'm seeing. Typically, overriding nullability is a bad thing. It may be correct here. Would like a second opinion @a-ctor

**Review by MichaelKetting — CHANGES_REQUESTED** [review]
Requesting changes, at least for the commit history, possibly code, too.

File: Remotion/Core/Development.Analyzers/RMCORE0001_SafeContextAnalyzer.cs (hunk 1 above)
Comment (Frazerus) [reply Remotion/Core/Development.Analyzers/RMCORE0001_SafeContextAnalyzer.cs:20]: This has always been with `.` at the end here, I do not remember a discussion about this however.

Comment (MichaelKetting) [reply Remotion/Core/Development.Analyzers/RMCORE0001_SafeContextAnalyzer.cs:20]: I just found a current example and yeah, Microsoft doesn't place a `.` at the end. It probably dates back to Marcel. 
Question for @a-ctor  and @Frazerus change this now, later or not at all?
Question for @KuJo7 : is there a ready-made assertion in the Microsoft-Library which forced this change or did you simply notice it and adapt it to be consistent with Microsoft?

Comment (a-ctor) [reply Remotion/Core/Development.Analyzers/RMCORE0001_SafeContextAnalyzer.cs:20]: With a newer version of roslyn a check for `<EnforceExtendedAnalyzerRules>true</EnforceExtendedAnalyzerRules>` was introduced and this is what causes all those changes now as it has an error severity. I would do the changes now as they don't seem to be too much work.

File: Remotion/Web/Development.Analyzers/AnalyzerReleases.Unshipped.md (hunk 2 above)
Comment (a-ctor) [reply Remotion/Web/Development.Analyzers/AnalyzerReleases.Unshipped.md:1]: We did not need it because compile did not complain back then. Now it does so we should add them as is normal with all analyzer projects. To this day, I don't know if they have any technical relevance or are just here for documentation purposes...

File: Remotion/Core/Development.Analyzers/Core.Development.Analyzers.csproj (hunk 3 above)
Comment (a-ctor) [reply Remotion/Core/Development.Analyzers/Core.Development.Analyzers.csproj:10]: As the name suggest it enforces analyzer best practices that are not enforced before. With the update the analyzer class had an error that complained that this property should be set. If we don't want to do all the work now we might be able to set this to `false`, which might disable the initial error as well.

File: Remotion/Web/Development.Analyzers/WxeResourcePageStepAnalyzer.cs (hunk 4 above)
Comment (a-ctor) [reply Remotion/Web/Development.Analyzers/WxeResourcePageStepAnalyzer.cs:96]: We should probably use `?.` instead as that won't fail and the equality still works in our favor.

File: Remotion/Core/Development.Analyzers/Core.Development.Analyzers.csproj (hunk 3 above)
Comment (MichaelKetting) [reply Remotion/Core/Development.Analyzers/Core.Development.Analyzers.csproj:10]: Good. 
@KuJo7 We should do a separate commit that fixes the trailing periods and introduces the build property and probably the unshipped.md file and document in the comment message that the change was required due to stronger enforcement of rules. Otherwise, we don't know why it was done when we look at the code history again. We accept that the build wouldn't work for this first revision without the changes since it's only in the feature branch. in theory, we could do the fix-commit first and then the upgrade-commit, but I don't think it's worth it.

File: Remotion/Core/Development.Analyzers/RMCORE0001_SafeContextAnalyzer.cs (hunk 1 above)
Comment (MichaelKetting) [reply Remotion/Core/Development.Analyzers/RMCORE0001_SafeContextAnalyzer.cs:20]: See note in Core.Development.Analyzers.csproj

**Review by MichaelKetting — APPROVED** [review]
